delivery_postlogistics/models/stock_picking.py

```diff
@@ -155,6 +155,11 @@ def write_tracking_number_label(self, label_result, packages):
         zpl_patch_string = self.carrier_id.zpl_patch_string
 
         labels = []
+
+        # It could happen that no successful label has been returned by the API
+        if not label_result:
+            return labels
```

Comment:
I would not place this check in `write_tracking_number_label` but perhaps in `_generate_postlogistics_label` where `success_label_results` (and `failed_label_results`) are initialized => https://github.com/OCA/delivery-carrier/blob/4a66d8ed7207468e007b411ea28914dabe86784a/delivery_postlogistics/models/stock_picking.py#L205-L217

Furthermore it would be interesting to know why the label hasn't been generated? Is the `failed_label_results` empty too? On the link above you can see that if `failed_label_results` is not empty, a rollback is performed but no exception is raised, so the flow continues as if nothing happened: why it has been done like this? Is it on purpose?
ping @phuctranerp @jbaudoux

Reply:
The reason why I placed this check here was to avoid code like this:
```python3
labels = []
if success_label_results:
    labels = self.write_tracking_number_label(success_label_results, packages)
```

Reply:
But you say that `success_label_results` is empty, ok, what about `failed_label_results`? Because `write_tracking_number_label` is called after the rollback, and maybe instead of a rollback we simply need to raise an exception?

Reply:
The exception is raised after, with the error message if `failed_label_results` is not empty, whether `success_label_result` is empty or not.
We just miss something, to disable actions on success labels in such case.

Reply:
Yes this is also an issue we encountered with @santostelmo the other day (related to this fix: https://github.com/OCA/delivery-carrier/pull/371): we were losing the packages generated in the same transaction with this rollback, even the good ones.

Ok then for your fix!